content/blog/2018-09-14-smapr.md

```diff
@@ -0,0 +1,637 @@
+- The empirical CDFs provide a mapping from the range of data to the interval from zero to one, so that we can visualize how any particular amount of soil moisture or precipitation compares to the full distribution of values for each pixel. Essentially, when the empirical CDF gives a value close to 0, that's a low value (a small fraction of values are less than or equal to it). If the empirical CDF gives a value close to one, that's a high value (a large fraction of values are less than or equal to it).
+
+Let's see what the empirical CDF values look like on a map:
+
+
+```r
+sm_ecdf %>%
+  filter(date > as.Date('2018-01-01')) %>%
+  ggplot(aes(x=x, y=y, fill=ecdf)) +  
+  geom_raster() +
+  scale_fill_viridis_c('Soil moisture\nempirical CDF',
+                       option = 'B', direction = -1) +
+  facet_wrap(~date, nrow = 7) +
+  theme_minimal() +
+  theme(axis.text = element_blank()) +
+  geom_sf(data = study_area_sf, inherit.aes = FALSE, fill = NA, size = .2) +
+  xlab('') +
+  ylab('')
+```
+
+![](/img/blog-images/2018-09-14-smapr/soil-moisture-raster-cdf-1.png)<!-- -->
+
+
+
+```r
+precip %>%
+  group_by(x, y) %>%
+  mutate(ecdf = ecdf(pr)(pr)) %>%
+  filter(date > as.Date('2018-01-01')) %>%
+  ggplot(aes(x=x, y=y, fill=ecdf)) +  
+  geom_raster() +
+  scale_fill_viridis_c('Precipitation\nempirical CDF',
+                       option = 'B',
+                       direction = -1) +
+  facet_wrap(~date, nrow = 7) +
+  theme_minimal() +
+  theme(axis.text = element_blank()) +
+  geom_sf(data = study_area_sf, inherit.aes = FALSE, fill = NA, size = .2) +
+  xlab('') +
+  ylab('')
+```
+
+![](/img/blog-images/2018-09-14-smapr/precip-raster-cdf-1.png)<!-- -->
+
+Across much of Somalia and Kenya, soils were exceptionally dry prior to March 2018 (light colors), and with the prolonged levels of high rain beginning in March, soils were much more saturated than usual through mid to late May (dark colors).
+The rain throughout April fell on wet soils, which are less able to absorb water, leading to overland flow and flooding.
+
+We can put this all together in a visualization that shows the evolution of soil moisture and precipitation:
+
+![](/img/blog-images/2018-09-14-smapr/joint-timeseries.gif)
+
+Here the soil moisture and precipitation values are mapped over time, with line plots below to show the median value (black line), interquartile range (dark grey ribbon), 10% and 90% quantiles (light grey ribbon).
+The exceptional nature of the flooding in 2018 shows up as a high and broad peak in soil moisture and precipitation, and the drought that preceded the flooding in 2017 is visible as a period of low rains and dry soil.
+
+
+### Want to contribute?
+
+If you're psyched on global soil moisture and want to contribute to smapr, there is room to develop support for more level 2 products (currently smapr supports the more processed level 3 science grade and level 4 enhanced value products primarily).
```

Comment:
For a novice reader (like me), can you indicate what level 2 vs level 3, 4 products are?

Reply:
Done! https://github.com/ropensci/roweb2/pull/277/commits/969c944c405ea49113fcd1912fc368ea650d93c1